### DISCUSSION TITLE
Filtering for rendering only specific entities

### DISCUSSION THREAD

**Discussion by MrJakobLaich** [skypjack/entt#1062] [General]
I've looked but not found anything related to this.

What I have / will have (currently refactoring to use enTT)
registry that stores position, render data, etc
screen window with camera that renders only the specific part the player can see (using sdl2, 2d top down tilemap, centered on player position)
each tile/chunk in the map stores the entity IDs of whatever is on top of it, the IDs are collected when rendering the tiles/chunks (the IDs would get passed to the enTT registry and then I'd only render these)

Then I would want to somehow filter all the entities inside my registry, without having to iterate over every single entity  that has a position and render component and if-checking if the entity would be inside the render area

So, my question because I can't find anything related to this:

1) am I supposed to just get all entities that have a position and renderdata component and just check each if it should be rendered?
2) is there some way to do registry.get everything that has position inside given ± filter area and has a render component -> filtering specific components by value and get a packed array that only contains the filtered (inside value range in my case) entities
3) is there a way to get only specific entities from a registry -> registry.get and I pass it a list of entities and then list what components they should have (get only the entities that have position and render component from the supplied list of IDs)

**Comment by skypjack** [comment]
> Then I would want to somehow filter all the entities inside my registry, without having to iterate over every single entity that has a position and render component and if-checking if the entity would be inside the render area

Flip the problem on its head then. Iterate the entities in the render area and attach a _render-me_ component to them. Then iterate the entities that have a position, render **and** _render-me_ component. Finally, clear the _render-me_ pool for the next iteration and you're done.

**Reply by MrJakobLaich** [reply]
I see, I like that idea, sounds simple and clean, thanks 👍
